### PR TITLE
fix: Don't show non-templates when asking chosing

### LINKED ISSUE #923: Ignore dotfiles when listing templates
Files starting with a `.` in their name should be ignored when listing available templates for issues and MRs:

```console
$ glab issue create
? Choose a template  [Use arrows to move, type to filter]
  .gitkeep
  template1
  template2
> template3
  Open a blank Issue
```

Possible code location: https://github.com/profclems/glab/blob/e866496799db83a5a5ef939ed78509f3775167c1/commands/cmdutils/cmdutils.go#L85-L88

### PR DESCRIPTION
**Pull request by zemzale**
## Description

In cases where users have non template files in the .gitlab/[template]/
folder they might will be picked up and displayed.

This fixes that by filtering out all the files that are hidden and don't
end in .md

## Related Issue

Resolves #923

## How Has This Been Tested?

Unit tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

### CODE REVIEW

**Comment by zemzale** [comment]
I don't know if this is the correct way how to setup testing for this, but it works 😄

**Review by profclems — CHANGES_REQUESTED** [review]
This looks good but I'm only concerned about the new test files in the main issue templates folder.

Maybe we could move them to `test/testdata/.gitlab` ?

**Comment by zemzale** [comment]
I moved the test files to the testdata directory. 

I had to override the git function for getting top-level directory, but this should be fine as long as we don't allow overriding this outside the tests.

**Comment by profclems** [comment]
> I moved the test files to the testdata directory.
> 
> I had to override the git function for getting top-level directory, but this should be fine as long as we don't allow overriding this outside the tests.

Sounds good.

File: commands/cmdutils/cmdutils_test.go [hunk 1]
```diff
@@ -1350,3 +1351,33 @@ func Test_ConfirmSubmission(t *testing.T) {
...
+	tests := []struct {
+		name          string
+		give          string
+		wantTemplates []string
+		wantErr       bool
+	}{
+		{
+			name:          "Get all the issues templates",
+			give:          "issue_templates",
+			wantTemplates: []string{"Feature Request", "Bug"},
```
Comment (profclems) [inline commands/cmdutils/cmdutils_test.go]: I think it should be `[]string{"Bug", "Feature Request"}` instead.

**Review by profclems — CHANGES_REQUESTED** [review]
One last fix for the tests

File: commands/cmdutils/cmdutils_test.go (hunk 1 above)
Comment (zemzale) [reply commands/cmdutils/cmdutils_test.go]: So they show up in alphabetic order? I am just gonna sort the list before returning it.

**Review by profclems — APPROVED** [review]

**Pull request merged**